### PR TITLE
D8CORE-1828 Enable google tag manager

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Enable google tag manager
- ignore GTM container config entiies
- Use config override to disable GTM containers on non-prod environments.
- Allow config changes with config readonly https://github.com/SU-SWS/acsf-cardinalsites/pull/173

# Need Review By (Date)
- 7/22

# Urgency
- medium

# Steps to Test
1. `composer require su-sws/stanford_profile:dev-D8CORE-1828`
1. do a fresh install or `blt drupal:update`
1. add a container `/admin/config/system/google-tag`
1. verify its disabled by viewing the page source when incognito or using `drush cget google_tag.container.[container_name] --include-overridden`

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: config/sync/core.extension.yml [hunk 1]
```diff
@@ -33,6 +33,7 @@ module:
   datetime: 0
   datetime_range: 0
   default_content: 0
+  diff: 0
```
Comment (sherakama) [inline config/sync/core.extension.yml:36]: Is diff needed or is this just a nice to have?

File: composer.json [hunk 2]
```diff
@@ -76,13 +76,10 @@
         "drupal/address": "~1.2",
         "drupal/admin_toolbar": "^2.2",
         "drupal/allowed_formats": "~1.1",
-        "drupal/better_normalizers": "^1.0-beta3",
-        "drupal/block_blacklist": "~1.0",
```
Comment (sherakama) [inline composer.json:80]: We use this. Is it somewhere else?

File: composer.json [hunk 3]
```diff
@@ -113,18 +110,15 @@
         "drupal/menu_admin_per_menu": "^1.0",
         "drupal/menu_block": "dev-1.x#6a2e5a8e83f0b8e26546aebf198d0b66ed4747c8",
         "drupal/menu_link_weight": "^1.0-beta3",
-        "drupal/menu_position": "~1.0@alpha",
```
Comment (sherakama) [inline composer.json:116]: We are going to need this for the view pages but that can be a different PR.

File: config/sync/google_tag.settings.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+uri: 'public:/'
```
Comment (sherakama) [inline config/sync/google_tag.settings.yml:1]: Shouldn't this be `public://`?

File: config/sync/core.extension.yml (hunk 1 above)
Comment (pookmish) [reply config/sync/core.extension.yml:36]: just nice to have. i noticed we had the module in composer so i figured we could enable it.

File: composer.json (hunk 2 above)
Comment (pookmish) [reply composer.json:80]: its not in core.extension.yml so i don think we're using it.

File: config/sync/google_tag.settings.yml (hunk 4 above)
Comment (pookmish) [reply config/sync/google_tag.settings.yml:1]: I was a little confused by that too. but this is the default settings. i'm not 100% sure why its only 1 slash

File: composer.json (hunk 2 above)
Comment (sherakama) [reply composer.json:80]: I am totally thinking of layout builder restrictions. 

All good.
